framework/auth/views.py

```diff
@@ -637,3 +714,114 @@ def resend_confirmation_post(auth):
 
     # Don't go anywhere
     return {'form': form}
+
+
+def external_login_email_get():
+    """
+    Landing view for first-time oauth-login user to enter their email address.
+    HTTP Method: GET
+    """
+
+    form = ResendConfirmationForm(request.form)
+    session = get_session()
+    if not session.is_external_first_login:
+        raise HTTPError(http.UNAUTHORIZED)
+
+    external_id_provider = session.data['auth_user_external_id_provider']
+
+    return {
+        'form': form,
+        'external_id_provider': external_id_provider.upper()
```

Comment:
Agreed. `external_id_provider` used to be all lower-case and this `.upper()` is only for **ORCID** . With the new PR, it becomes what they should be in default settings for different external identity providers. There are other places that I used `.upper()` and I will fix all.